### PR TITLE
Update integretion workflow compatible with CML 2.5.1

### PR DESCRIPTION
**Pull request by ashwini-mhatre**

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -41,17 +41,12 @@ jobs:
       fail-fast: false
       matrix:
         ansible-version:
-          # - stable-2.9
-          - stable-2.13
-          # - stable-2.14
+          - stable-2.14
```
Comment (NilashishC) [inline .github/workflows/integration.yml:44]: Can we bump this to stable-2.16? stable-2.14 is going to be EoL soon.

**Review by NilashishC — CHANGES_REQUESTED** [review]

File: .github/actions/build_install_collection/action.yml [hunk 2]
```diff
@@ -5,6 +5,8 @@ inputs:
   install_python_dependencies:
     description: "Install collection python dependencies"
     required: true
+  python_version:
+    default: "3.9"
```
Comment (abikouo) [inline .github/actions/build_install_collection/action.yml:9]: This is a breaking change as it is supposed all workflows currently using this action should have `python3.9` installed in their environment

File: .github/actions/build_install_collection/action.yml [hunk 3]
```diff
@@ -67,7 +69,7 @@ runs:
     - name: Install collection python requirements
       if: ${{ inputs.install_python_dependencies == 'true' }}
 
-      run: python3 -m pip install -r requirements.txt -r test-requirements.txt
+    - run: python${{ matrix.python-version }} -m pip install -r requirements.txt -r test-requirements.txt
```
Comment (abikouo) [inline .github/actions/build_install_collection/action.yml:72]: you should also install `bindep` using the same python version (at line 53)

**Review by abikouo — CHANGES_REQUESTED** [review]

File: .github/actions/build_install_collection/action.yml (hunk 2 above)
Comment (NilashishC) [reply .github/actions/build_install_collection/action.yml:9]: @abikouo Agreed. I think this option isn't even being used anywhere. We should just remove this?

**Review by NilashishC — APPROVED** [review]

File: .github/workflows/integration.yml [hunk 4]
```diff
@@ -112,24 +111,31 @@ jobs:
       - name: Build and install the collection
         uses: ansible-network/github_actions/.github/actions/build_install_collection@main
         with:
+          python_version: ${{ matrix.python-version }}
```
Comment (NilashishC) [inline .github/workflows/integration.yml:114]: We have removed this input, so this is no longer needed.

**Review by NilashishC — CHANGES_REQUESTED** [review]

**Comment by ashwini-mhatre** [comment]
@abikouo Addressed review comments .could you please have look?

**Comment by ashwini-mhatre** [comment]
@GomathiselviS @cidrblock could you please have look to this PR?

File: .github/actions/build_install_collection/action.yml [hunk 5]
```diff
@@ -48,7 +48,12 @@ runs:
       working-directory: ${{ inputs.source_path }}
 
     - name: Install bindep from pypi
-      run: sudo python3 -m pip install bindep
+      run: |
+        if ${{ matrix.python-version != ''}}; then
+          python${{ matrix.python-version }} -m pip install bindep
```
Comment (NilashishC) [inline .github/actions/build_install_collection/action.yml:53]: Can't we use `python${{ matrix.python-version || '3' }}` here?

**Review by NilashishC — APPROVED** [review]

**Pull request merged**